File: array/168.py
from typing import List
import math


def containsNearbyDuplicate(nums: List[int], k: int) -> bool:
    stats = {}

    for index, value in enumerate(nums):
        if value in stats and abs(stats[value] - index) <= k:
            return True
        else:
            stats[value] = index

    return False


print(containsNearbyDuplicate([1, 2, 3, 1], 3))

File: array/168.cpp
#include <bits/stdc++.h>

using namespace std;

bool containsNearbyDuplicate(vector<int>& nums, int k) {
    unordered_map<int, int> um;

    for(int i = 0; i < nums.size(); ++i) {
        if(um.count(nums[i]) > 0 && abs(um[nums[i]] - i) <= k)
            return true;
        else {
            um[nums[i]] = i;
        }
    }

    return false;
}

int main() {
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    vector<int> nums = {1,2,3,1,2,3};
    int k = 2;

    cout<<boolalpha<<containsNearbyDuplicate(nums, k);

    return 0;
}